UnityProject/Assets/Scripts/Items/Weapons/Explosive.cs

```diff
@@ -85,17 +88,32 @@ private void Detonate()
 			if (isServer)
 			{
 				// Get data before despawning
-				var explosionMatrix = registerItem.Matrix;
 				var worldPos = objectBehaviour.AssumedWorldPositionServer();
 
 				// Despawn the explosive
 				_ = Despawn.ServerSingle(gameObject);
+				Explosion.StartExplosion(worldPos, explosiveStrength);
+			}
+
+		}
 
-				// Explosion here
-				var explosionGO = Instantiate(explosionPrefab, explosionMatrix.transform);
-				explosionGO.transform.position = worldPos;
-				explosionGO.Explode(explosionMatrix);
+		[Server]
+		private void CmdAttachExplosive(GameObject target, Vector2 targetPostion)
```

Comment:
so,  CMD how do you check that  the interaction is valid?

Reply:
and not just made up by a hacked client to cheat

Reply:
It's set to `[server]` only so i don't know how a client can trigger it

Reply:
change the name then  since CMD is reserved for cmds